### DISCUSSION TITLE
I configured a filter for the producer but it executes 2 times for every request and I don't know if this is reasonable

### DISCUSSION THREAD

**Discussion by xdxTao** [apache/dubbo#14545] [Question]
### Pre-check

- [X] I am sure that all the content I provide is in English.

### Apache Dubbo Component

Java SDK (apache/dubbo)

### Details

I just built a demo, so the code is very simple.

The filter configuration is as follows and it can be executed normally.

The problem is that it is executed twice every time

<br/>

<img width="1799" alt="image" src="https://github.com/user-attachments/assets/e6027900-f418-4557-8735-46a4e9b69cc9">

<br/><br/>

I found the execution points twice through debugging.

<img width="1497" alt="image" src="https://github.com/user-attachments/assets/44e4bf9c-8227-47ac-8047-47f02a6189d0">
<img width="1477" alt="image" src="https://github.com/user-attachments/assets/d3b06cee-6647-4ee8-be8a-1ab77cfb87d8">

<br/><br/>

Conventional understanding is that filter should only be executed once.

What if I only want to call the filter once on the producer?

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by AlbumenJ** [comment]
![image](https://github.com/user-attachments/assets/89fb716f-7456-42cc-90c6-0dcbdd07c8f2)
This is the provider side

![image](https://github.com/user-attachments/assets/ba355ad7-aadc-4b9f-b40c-72bb6dbf378a)
This is the consumer side

Have you ever configured `dubbo.consumer.filter=xxx`?

**Reply by xdxTao** [reply]
There is no `dubbo.consumer.filter=xxx` in my configuration file.

<br/>

I found the reason is because this was introduced in my POM file.

<img width="1610" alt="image" src="https://github.com/user-attachments/assets/a6342f38-9373-46c1-8ff4-28e099278a0c">


<br/><br/><br/>

If I comment it out, it won't be executed twice. But I don't understand why this happens @AlbumenJ
